File: java/eyes.playwright.java/src/main/java/com/applitools/eyes/playwright/universal/dto/AccessibilityRegionBySelector.java
package com.applitools.eyes.playwright.universal.dto;

import com.applitools.eyes.fluent.GetRegion;
import com.applitools.eyes.universal.dto.TAccessibilityRegion;
import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class AccessibilityRegionBySelector extends TAccessibilityRegion implements GetRegion {

    private Selector region;

    public Selector getRegion() {
        return region;
    }

    public void setRegion(Selector region) {
        this.region = region;
    }

}

File: dotnet/Eyes.Playwright.DotNet/Fluent/AccessibilityRegionBySelector.cs
using Applitools.Fluent;
using Applitools.Playwright.Universal.Dto;
using Applitools.Universal;
using Microsoft.Playwright;
using Refer = Applitools.Playwright.Universal.Refer;

namespace Applitools.Playwright.Fluent
{
    /// <summary>
    /// Used internally to represent a ref of an accessibility element
    /// </summary>
    public class AccessibilityRegionBySelector : Selector, IGetAccessibilityRegion, IGetAccessibilityRegionType, IPlaywrightReference<TAccessibilityRegion>
    {
        public AccessibilityRegionType AccessibilityRegionType { get; }

        public AccessibilityRegionBySelector(string selector, AccessibilityRegionType type)
            : base(selector)
        {
            AccessibilityRegionType = type;
        }

        public AccessibilityRegionBySelector(ILocator locator, AccessibilityRegionType type) : base(locator)
        {
            AccessibilityRegionType = type;
        }

        TAccessibilityRegion IGetAccessibilityRegion.ToRegion()
        {
            return new SelectorAccessibilityRegion()
            {
                Type = AccessibilityRegionType,
                Region = new RegionSelector
                {
                    //ElementId = base.ElementHandle
                }
            };
        }

        TAccessibilityRegion IPlaywrightReference<TAccessibilityRegion>.ToRegion(Reference root, Refer refer)
        {
            return new SelectorAccessibilityRegion()
            {
                Type = AccessibilityRegionType,
                Region = GetRegionSelector(root, refer)
            };
        }
    }
}